app/models/api/v1/report.rb

```diff
@@ -320,7 +322,17 @@ def self.submit_feedback(answer_feedback_hash)
     end
 
     return unless answer
-    answer.add_feedback(answer_feedback_hash)
+    if answer.respond_to? :add_feedback # the answer is a saveable
+      answer.add_feedback(answer_feedback_hash)
+    else # assume answer has feedback columns.
+      answer.update_attributes(answer_feedback_hash)
+    end
+    # All answers should delegate learner to their saveables...
+    # We need to update fields, because the answer is serialized in the report learner
+    if answer.respond_to?(:learner) && answer.learner
+      answer.learner.report_learner.update_fields()
```

Comment:
This is going to be pretty inefficient. If a teacher is rapidly submitting feedback on a large investigation. But I don't think we should fix it now. If it becomes a problem we'll see it as a slow request in New Relic.